<!DOCTYPE html>
<html>
 <head> 
  <title>MarvinSketch : 6.0</title> 
  <link rel="stylesheet" href="styles/site.css" type="text/css" /> 
  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" /> 
  <style>
#content {
    -moz-box-sizing: border-box;
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    clear: both;
    position: relative;
    background: #FFFFFF !important;
}

.aui-page-panel {
    background: #FFFFFF !important;
    border: 1px solid #ccc;
    border-left-width: 0;
    border-right-width: 0;
    -moz-box-sizing: border-box;
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    clear: both;
    display: block;
    margin: 20px 0 0 0;
    padding: 0;
    position: relative;
    width: 100%;
}
.message-content {
	border: solid 2px #fee796;
	background-color: #fefaea;
	border-radius: 10px;
	display: table;
	padding: 4px;
	margin: 4px 4px 0px 4px;
}
</style> 
 </head> 
 <body class="theme-default aui-theme-default"> 
  <div id="page"> 
   <div id="main" class="aui-page-panel"> 
    <div id="main-header"> 
     <h1 id="title-heading" class="pagetitle"> <span id="title-text"> MarvinSketch : 6.0 </span> </h1> 
    </div> 
    <div id="content" class="view"> 
     <div id="main-content" class="wiki-content group"> 
      <p></p>
      <div class="contentLayout2"> 
       <div class="columnLayout three-equal" data-layout="three-equal"> 
        <div class="cell normal" data-type="normal"> 
         <div class="innerCell"> 
          <div class="panel" style="border-width: 1px;">
           <div class="panelContent"> 
            <p><style type="text/css">/*<![CDATA[*/
div.rbtoc1439201040842 {padding: 0px;}
div.rbtoc1439201040842 ul {list-style: disc;margin-left: 0px;}
div.rbtoc1439201040842 li {margin-left: 0px;padding-left: 0px;}

/*]]>*/</style></p>
            <div class="toc-macro rbtoc1439201040842"> 
             <ul class="toc-indentation"> 
              <li><a href="https://docs.chemaxon.com/display/products/6.0#id-6.0-November25th,2013:Marvin6.0.6">November 25th, 2013: Marvin 6.0.6</a> 
               <ul class="toc-indentation"> 
                <li><a href="https://docs.chemaxon.com/display/products/6.0#id-6.0-Bugfixes">Bugfixes</a></li> 
               </ul> </li> 
              <li><a href="https://docs.chemaxon.com/display/products/6.0#id-6.0-August27th,2013:Marvin6.0.5">August 27th, 2013: Marvin 6.0.5</a> 
               <ul class="toc-indentation"> 
                <li><a href="https://docs.chemaxon.com/display/products/6.0#id-6.0-Bugfixes.1">Bugfixes</a></li> 
               </ul> </li> 
              <li><a href="https://docs.chemaxon.com/display/products/6.0#id-6.0-August6th,2013:Marvin6.0.4">August 6th, 2013: Marvin 6.0.4</a></li> 
              <li><a href="https://docs.chemaxon.com/display/products/6.0#id-6.0-July23th,2013:Marvin6.0.3">July 23th, 2013: Marvin 6.0.3</a></li> 
              <li><a href="https://docs.chemaxon.com/display/products/6.0#id-6.0-June25th,2013:Marvin6.0.2">June 25th, 2013: Marvin 6.0.2</a></li> 
              <li><a href="https://docs.chemaxon.com/display/products/6.0#id-6.0-June21st,2013:Marvin6.0.1">June 21st, 2013: Marvin 6.0.1</a> 
               <ul class="toc-indentation"> 
                <li><a href="https://docs.chemaxon.com/display/products/6.0#id-6.0-Newfeaturesandimprovements">New features and improvements</a></li> 
                <li><a href="https://docs.chemaxon.com/display/products/6.0#id-6.0-Bugfixes.2">Bugfixes</a></li> 
               </ul> </li> 
              <li><a href="https://docs.chemaxon.com/display/products/6.0#id-6.0-May17th,2013:Marvin6.0.0">May 17th, 2013: Marvin 6.0.0</a> 
               <ul class="toc-indentation"> 
                <li><a href="https://docs.chemaxon.com/display/products/6.0#id-6.0-Newfeaturesandimprovements.1">New features and improvements</a></li> 
                <li><a href="https://docs.chemaxon.com/display/products/6.0#id-6.0-Knownissue">Known issue</a></li> 
                <li><a href="https://docs.chemaxon.com/display/products/6.0#id-6.0-Bugfixes.3">Bugfixes</a></li> 
                <li><a href="https://docs.chemaxon.com/display/products/6.0#id-6.0-Deprecation">Deprecation</a></li> 
               </ul> </li> 
             </ul> 
            </div>
            <p></p> 
           </div>
          </div>
         </div> 
        </div> 
        <div class="cell normal" data-type="normal"> 
         <div class="innerCell"> 
          <p>&nbsp;</p>
         </div> 
        </div> 
        <div class="cell normal" data-type="normal"> 
         <div class="innerCell"> 
          <p>&nbsp;</p>
         </div> 
        </div> 
       </div> 
       <div class="columnLayout single" data-layout="single"> 
        <div class="cell normal" data-type="normal"> 
         <div class="innerCell"> 
          <h3 id="id-6.0-November25th,2013:Marvin6.0.6">November 25th, 2013: Marvin 6.0.6</h3>
          <p><a class="external-link" href="http://www.chemaxon.com/marvin/help/marvin4js/marvinjs_changes.html#v60060" rel="nofollow">Marvin for JavaScript</a> 6.0.6 history of changes</p>
          <h4 id="id-6.0-Bugfixes">Bugfixes</h4>
          <ul>
           <li><strong>Applet</strong>
            <ul>
             <li>The mol property change event was not fired when an empty molecule was set initially for the MarvinSketch applet. This could cause misbehaviour in cases where the client JavaScript code relied on this marker to know when the applet was initialized.</li>
            </ul></li>
          </ul>
          <h3 id="id-6.0-August27th,2013:Marvin6.0.5">August 27th, 2013: Marvin 6.0.5</h3>
          <p><a class="external-link" href="http://www.chemaxon.com/marvin/help/marvin4js/marvinjs_changes.html#v60040" rel="nofollow">Marvin for JavaScript</a> 6.0.5 history of changes</p>
          <h4 id="id-6.0-Bugfixes.1">Bugfixes</h4>
          <ul>
           <li><strong>MarvinView GUI</strong>
            <ul>
             <li>Opening a structure for editing in MarvinView was opened with 0% zoom factor in the MarvinSketch window, when the structure contained an R-group.</li>
            </ul></li>
          </ul>
          <h3 id="id-6.0-August6th,2013:Marvin6.0.4">August 6th, 2013: Marvin 6.0.4</h3>
          <p><a class="external-link" href="http://www.chemaxon.com/marvin/help/marvin4js/marvinjs_changes.html#v60040" rel="nofollow">Marvin for JavaScript</a> 6.0.4 history of changes</p>
          <ul>
           <li><strong>MarvinSpace</strong>
            <ul>
             <li>Transferring 3D molecules from MarvinSpace to MarvinSketch resulted in 2D structures in the sketcher. <a class="external-link" href="https://www.chemaxon.com/forum/ftopic10900.html" rel="nofollow">Forum topic</a></li>
            </ul></li>
          </ul>
          <h3 id="id-6.0-July23th,2013:Marvin6.0.3">July 23th, 2013: Marvin 6.0.3</h3>
          <p><a class="external-link" href="http://www.chemaxon.com/marvin/help/marvin4js/marvinjs_changes.html#v60030" rel="nofollow">Marvin for JavaScript</a> 6.0.3 history of changes</p>
          <p>No changes.</p>
          <h3 id="id-6.0-June25th,2013:Marvin6.0.2">June 25th, 2013: Marvin 6.0.2</h3>
          <p><a class="external-link" href="http://www.chemaxon.com/marvin/help/marvin4js/marvinjs_changes.html#v60020" rel="nofollow">Marvin for JavaScript</a> 6.0.2 history of changes</p>
          <p>No changes.</p>
          <h3 id="id-6.0-June21st,2013:Marvin6.0.1">June 21st, 2013: Marvin 6.0.1</h3>
          <p><a class="external-link" href="http://www.chemaxon.com/marvin/help/marvin4js/marvinjs_changes.html#v60010" rel="nofollow">Marvin for JavaScript</a> 6.0.1 history of changes</p>
          <h4 id="id-6.0-Newfeaturesandimprovements">New features and improvements</h4>
          <ul>
           <li><strong>Image I/O</strong>
            <ul>
             <li>The following file extensions were removed from the &quot;Files of Type&quot; list on the Import Image dialog: *.ps, *.pdf, and &quot;All Files&quot;. PDF files can be imported through the &quot;File &gt; Open&quot; menu.</li>
            </ul></li>
           <li><strong>Calculations</strong>
            <ul>
             <li><strong>Services</strong>
              <ul>
               <li>SOAP webservices:
                <ul>
                 <li>SOAP version 1.2 webservices are supported.</li>
                </ul></li>
              </ul></li>
            </ul></li>
          </ul>
          <h4 id="id-6.0-Bugfixes.2">Bugfixes</h4>
          <ul>
           <li><strong>Painting</strong>
            <ul>
             <li>R/S stereo labels were not displayed when there was an R-group on the canvas.</li>
             <li>S-groups, created from structures containing a position variation bond, retained the grey patch of the position variation bond when the group was contracted.</li>
            </ul></li>
           <li><strong>Editing</strong>
            <ul>
             <li>An atom of an expanded S-group could not be replaced with an S-group correctly</li>
            </ul></li>
           <li><strong>Clipboard handling</strong>
            <ul>
             <li>On Mac OS X, copy-paste (from MarvinSketch to, <em>e.g.</em>, TextEdit) did not transfer the atom colors. <a class="external-link" href="https://www.chemaxon.com/forum/ftopic10144.html" rel="nofollow">Forum topic</a></li>
            </ul></li>
           <li><strong>Graphical object handling</strong>
            <ul>
             <li>Formatting the head of a graphical arrow to a half-left arrow using the Document Style dialog crashed the display of the arrow.</li>
            </ul></li>
           <li><strong>Import/Export</strong>
            <ul>
             <li><strong>MRV/CML</strong>
              <ul>
               <li>MRV and CML export failed with the woodstock xml processor. <a class="external-link" href="https://www.chemaxon.com/forum/ftopic10499.html" rel="nofollow">Forum topic</a></li>
              </ul></li>
             <li><strong>InChi/InChiKey</strong>
              <ul>
               <li>InChi export did not return standard InChi when the absolute stereo was not set in a non-chiral molecule. <a class="external-link" href="https://www.chemaxon.com/forum/ftopic10424.html" rel="nofollow">Forum topic</a></li>
              </ul></li>
             <li><strong>Document to Structure</strong>
              <ul>
               <li>Documents with many consecutive names would lead to extremely long processing time (appeared as hanging, unless reaching the timeout). <a class="external-link" href="https://www.chemaxon.com/forum/ftopic10643.html" rel="nofollow">Forum topic</a></li>
              </ul></li>
            </ul></li>
           <li><strong>Molecule Representation</strong>
            <ul>
             <li><strong>Implicit/Explicit Hydrogen Conversion</strong>
              <ul>
               <li>Adding explicit hydrogen to an atom being part of a Multicenter S-group added the hydrogen to the S-group as well. <a class="external-link" href="https://www.chemaxon.com/forum/ftopic10533.html" rel="nofollow">Forum topic</a></li>
              </ul></li>
            </ul></li>
           <li><strong>Clean 2D</strong>
            <ul>
             <li>Stereo clean created non-IUPAC standard depiction of certain tetrahedral configurations. <a class="external-link" href="https://www.chemaxon.com/forum/ftopic10614.html" rel="nofollow">Forum topic</a></li>
            </ul></li>
           <li><strong>Data Transfer</strong>
            <ul>
             <li><strong>OLE</strong>
              <ul>
               <li>Redirecting ChemDraw OLE objects did not work.</li>
              </ul></li>
            </ul></li>
           <li><strong>Applet</strong>
            <ul>
             <li>Applet parameter <code>extraBonds</code> did not work therefore the bond type menu appearance could not be changed. <a class="external-link" href="https://www.chemaxon.com/forum/ftopic10418.html" rel="nofollow">Forum topic</a></li>
            </ul></li>
           <li><strong>Image I/O</strong>
            <ul>
             <li>The xml header information was always included in the generated SVG content during image export. SVG export now has an option &quot;headless&quot; to exclude this information. <a class="external-link" href="http://www.chemaxon.com/marvin/help/formats/svg-doc.html" rel="nofollow">Documentation</a>, <a class="external-link" href="https://www.chemaxon.com/forum/ftopic10002.html" rel="nofollow">Forum topic</a></li>
            </ul></li>
           <li><strong>Calculations</strong>
            <ul>
             <li><strong>Protonation (pKa, Major Microspecies, Isoelectric Point)</strong>
              <ul>
               <li>&quot;Keep explicit hydrogen&quot; option caused exception during pKa calculation.</li>
              </ul></li>
             <li><strong>NMR (HNMR, CNMR Prediction, ...)</strong>
              <ul>
               <li>When the mouse cursor was moved over an NMR peak on the chart, the relevant atom was not highlighted on NMR predictor molecule preview panel.</li>
              </ul></li>
             <li><strong>Services</strong>
              <ul>
               <li>SOAP webservices:
                <ul>
                 <li>Parameters did not load from WSDL properly in SOAP version 1.1 webservices.</li>
                </ul></li>
              </ul></li>
            </ul></li>
          </ul>
          <h3 id="id-6.0-May17th,2013:Marvin6.0.0">May 17th, 2013: Marvin 6.0.0</h3>
          <h4 id="id-6.0-Newfeaturesandimprovements.1">New features and improvements</h4>
          <ul>
           <li><strong>MarvinSketch GUI</strong>
            <ul>
             <li>The default tool in MarvinSketch is Rectangle Selection instead of Single Bond.</li>
             <li>MarvinSketch GUI has been refreshed: Icon set has been changed in MarvinSketch. Menu items have been rearranged. Several dialogs, tooltips, menus has been renamed.</li>
             <li>A new Marvin configuration has been added to MarvinSketch. Already existing Marvin configurations have been renamed. <a class="external-link" href="http://www.chemaxon.com/marvin/help/sketch/gui/configurations.html" rel="nofollow">Documentation</a></li>
             <li>MarvinSketch now starts with the native &quot;Look &amp; Feel&quot; on Windows and on Mac OS X.</li>
            </ul></li>
           <li><strong>MarvinSketch Dialog</strong>
            <ul>
             <li>At the first launch of MarvinSketch after installation, a dialog asks the user to select the desired skin for the GUI configuration. <a class="external-link" href="http://www.chemaxon.com/marvin/help/sketch/gui/configurations.html" rel="nofollow">Documentation</a></li>
             <li>The Open dialog has been modified. Selecting molecules to load from a multistructure file happens now directly on the Open dialog.</li>
             <li>New dialog, called &quot;Insert file&quot;, has been introduced into MarvinSketch for adding structures from new files to the canvas contents. <a class="external-link" href="http://www.chemaxon.com/marvin/help/sketch/sketch-basic.html#open-molecule" rel="nofollow">Documentation</a></li>
             <li>The Save dialog has been extended with an Advanced panel to set saving options.</li>
            </ul></li>
           <li><strong>Painting</strong>
            <ul>
             <li>Drawing quality has been improved: join of single bonds has been refined.</li>
             <li>Superatom S-group attachment points are denoted by numbers instead of asterisks.</li>
            </ul></li>
           <li><strong>Editing</strong>
            <ul>
             <li>MarvinSketch GUI can handle unlimited number of attachment points on Superatom S-groups. There are new menu options for adding and removing attachment points in the Atom menu and in the contextual menu. <a class="external-link" href="http://www.chemaxon.com/marvin/help/sketch/sketch-basic.html#Sgoups_to_template" rel="nofollow">Documentation</a></li>
             <li>Bond length can be scaled. Bond length and size of other objects can be set by dragging the displayed corners of the bounding rectangle. Scaling is also available on the Document Style dialog. Scaling does not work when the structure has been rotated in 3D mode (with F7). <a class="external-link" href="http://www.chemaxon.com/marvin/help/sketch/sketc-basic.html#scalling" rel="nofollow">Documentation</a></li>
            </ul></li>
           <li><strong>Load/Save handling in UI</strong>
            <ul>
             <li>New loading mechanism has been introduced in MarvinSketch. New menu option, Insert file, has been added to the File menu; it appends the contents of the chosen file to the structure on the canvas. <a class="external-link" href="http://www.chemaxon.com/marvin/help/sketch/sketch-basic.html#open-molecule" rel="nofollow">Documentation</a></li>
             <li>New saving mechanism has been introduced in MarvinSketch. <a class="external-link" href="http://www.chemaxon.com/marvin/help/sketch/sketch-basic.html#save2molformat" rel="nofollow">Documentation</a></li>
             <li>New save mechanism has been introduced for MarvinView. <a class="external-link" href="http://www.chemaxon.com/marvin/help/view/view-basic.html" rel="nofollow">Documentation</a></li>
             <li>Reactions and structures containing R-group definitions cannot be saved either in separate files or separately in the same file. <a class="external-link" href="http://www.chemaxon.com/marvin/help/sketch/sketch-basic.html#save2molformat" rel="nofollow">Documentation</a></li>
            </ul></li>
           <li><strong>Clipboard handling</strong>
            <ul>
             <li>Images of chemical structures can be imported from the clipboard via the configured image import service or locally installed OSRA.</li>
            </ul></li>
           <li><strong>Template Library</strong>
            <ul>
             <li>New template set, Organometallics, has been added to the Template Library.</li>
             <li>New amino acid templates have been added (e.g., pyrrolysine, selenocysteine, &quot;any/unknown/undetermined aminoacid&quot;).</li>
             <li>The following amino acid templates have three attachment points: arginine, cysteine, histidine, lysine, selenocysteine, serine, threonine, tryptophan, tyrosine.</li>
            </ul></li>
           <li><strong>Integration</strong>
            <ul>
             <li>The new Automapper has been introduced in MarvinSketch.</li>
            </ul></li>
           <li><strong>MarvinSpace</strong>
            <ul>
             <li>Copy and Paste functions have been added to MarvinSpace.</li>
             <li>3D editor: Molecules can be edited in 2D in MarvinSketch, and aligned to the original 3D molecule.</li>
            </ul></li>
           <li><strong>Import/Export</strong>
            <ul>
             <li><strong>MRV/CML</strong>
              <ul>
               <li>Import and export of more than two Superatom S-group attachment points are supported.</li>
              </ul></li>
             <li><strong> MRV schema </strong>
              <ul>
               <li>New element, AttachmentPointArray, is introduced in format MRV to describe the new Superatom S-group attachment point representation. <a class="external-link" href="http://www.chemaxon.com/marvin/help/formats/schema/mrvSchema_6_0_0.xsd" rel="nofollow">MRV Schema</a>, <a class="external-link" href="http://www.chemaxon.com/marvin/help/formats/mrv-doc.html" rel="nofollow">Documentation</a></li>
              </ul></li>
             <li><strong>MOL, SDF, RXN, RDF</strong>
              <ul>
               <li>The import of template based MDL MOL files is introduced.</li>
               <li>Import and export of more than two Superatom S-group attachment points are supported from MDL MOL based formats.</li>
              </ul></li>
             <li><strong>SKC</strong>
              <ul>
               <li>Import and export of more than two Superatom S-group attachment points are supported. Superatom S-groups having at least one crossing bond are exported in their expanded form.</li>
              </ul></li>
             <li><strong>CDX/CDXML</strong>
              <ul>
               <li>Import and export of more than two Superatom S-group attachments point are supported.</li>
               <li>Curved lines can be imported from CDX files.</li>
              </ul></li>
             <li><strong>CXSMILES/CXSMARTS</strong>
              <ul>
               <li>Generic, Polymer, and Mixture type S-groups are stored in CXSMILES/CXSMARTS format. <a class="external-link" href="http://www.chemaxon.com/marvin/help/formats/cxsmiles-doc.html" rel="nofollow">Documentation</a></li>
              </ul></li>
             <li><strong>SMILES/SMARTS</strong>
              <ul>
               <li>In SMILES import, radicals are not added anymore to atoms with atomic number above chlorine, except for bromine and iodine. If implicit hydrogen can not be added to or removed from the atom (as it is in brackets in the SMILES definition), then a valence property is set to correct its valence if necessary. <a class="external-link" href="http://www.chemaxon.com/marvin/help/formats/smiles-doc.html" rel="nofollow">Documentation</a></li>
              </ul></li>
             <li><strong>Peptide</strong>
              <ul>
               <li>Amino acids: Cysteine, Aspartic acid, Glutamic acid, Histidine, Lysine, Asparagine, Glutamine, Arginine, Serine, Threonine, Selenocysteine, Tryptophane, and Tyrosine are imported with three attachment points from peptide format.</li>
               <li>New 1-Letter and 3-letter amino acid codes were added: U and Sec for selenocysteine; O and Pyl for pyrrolysine; X and Xaa for &quot;any/unknown/undetermined amino acid&quot;; B and Asx for &quot;Aspartic acid or Asparagine&quot;; Z and Glx for &quot;Glutamic acid or Glutamine&quot;; J and Xle for &quot;Leucine or Isoleucine&quot;.</li>
              </ul></li>
             <li><strong>FASTA</strong>
              <ul>
               <li>Peptide, DNA, and RNA import from FASTA format has been introduced. <a class="external-link" href="http://www.chemaxon.com/marvin/help/formats/fasta-doc.html" rel="nofollow">Documentation</a></li>
              </ul></li>
             <li><strong>Name to Structure (n2s)</strong>
              <ul>
               <li>Custom name-to-structure webservices can now be configured to receive the name in the middle of the URL, not just at the end, by using the [NAME] marker in the configuration URL.</li>
               <li>More forms of ranged generic names are supported, such as &quot;-C<sub>1-6</sub>alkyl&quot;. These are found extensively as part of markush structures in patents.</li>
              </ul></li>
             <li><strong>Document to Structure (d2s)</strong>
              <ul>
               <li>Text document processing has become faster (around 20% speedup for typical texts with a small proportion of chemical names).</li>
               <li>The context (surrounding text) in which names appear in the document is now available as a property of the returned structures.</li>
               <li>The <code>osraTimeout</code> option has been added to configure the maximum time to wait for OSRA to convert an image to structure (the default is 20 seconds).</li>
              </ul></li>
             <li><strong>Structure to Name(s2n)</strong>
              <ul>
               <li>The new format option &quot;cas#&quot; allows converting structures to their CAS Registry Number using a public webservice.</li>
              </ul></li>
            </ul></li>
           <li><strong>Molecule Representation</strong>
            <ul>
             <li><strong>S-groups </strong>
              <ul>
               <li>Any number of attachment points can be added to Superatom S-groups.</li>
               <li>S-group attachment point information can be set only to atoms being part of a Superatom S-group.</li>
              </ul></li>
             <li><strong>Valence Check</strong>
              <ul>
               <li>Valence check does not deal with free attachment points.</li>
               <li>New valence check option is introduced to allow/deny traditional forms of pentavalent Nitrogen.</li>
              </ul></li>
            </ul></li>
           <li><strong>Calculations</strong>
            <ul>
             <li><strong>Protonation (pKa, Major Microspecies, Isoelectric Point)</strong>
              <ul>
               <li>log[%] vs. pH distribution chart is introduced in case of pKa calculation. <a class="external-link" href="http://www.chemaxon.com/marvin/help/calculations/protonation.html" rel="nofollow">Documentation</a></li>
               <li>Major miscrospecies calculation is supported for molecules with coordinate bonds. <a class="external-link" href="https://www.chemaxon.com/forum/viewpost47864.html" rel="nofollow">Forum topic</a></li>
              </ul></li>
             <li><strong>NMR (HNMR, CNMR Prediction, ...)</strong>
              <ul>
               <li>NMR predictor considers novel coupling types.</li>
               <li>Updated trainingset has improved the NMR shift prediction accuracy in both 1H and 13C NMR predictors.</li>
               <li>1H NMR calculation has become faster as no 3D conformer generation is applied.</li>
              </ul></li>
             <li><strong>Geometry (Topology Analysis, Geometry, PSA, MSA)</strong>
              <ul>
               <li>Setting any ChemAxon supported aromatization method (general, basic, loose) is available in TopologyAnalyserPlugin which affects the result of many topology analyser calculations (e.g., aromatic atom/bond/ring count).</li>
              </ul></li>
             <li><strong>Chemical Terms</strong>
              <ul>
               <li>New <code>sortableFormula()</code> function is introduced. The generated sortable formula can be used to sort formulas in alphabetical order. <a class="external-link" href="http://www.chemaxon.com/marvin/help/chemicalterms/EvaluatorFunctions.html#sortableformula" rel="nofollow">Documentation</a></li>
              </ul></li>
            </ul></li>
           <li><strong>cxcalc</strong>
            <ul>
             <li>pKa training library can be used without relying on user home. Use option <code>--correctionlibrarypath</code> to read custom training file.</li>
             <li>New <code>sortableFormula</code> option is introduced. The generated sortable formula can be used to sort formulas in alphabetical order.</li>
            </ul></li>
           <li><strong>cxtrain</strong>
            <ul>
             <li>New option, <code>-p</code>, has been introduced to save the training result to custom path.</li>
            </ul></li>
           <li><strong>Automapper</strong>
            <ul>
             <li>New Automapper tool is introduced. It maps reactions more accurately. <a class="external-link" href="http://www.chemaxon.com/marvin/help/automapper/automapper.html" rel="nofollow">Documentation</a></li>
             <li>Automapper tool requires Standardizer license.</li>
             <li>API documentation: <a class="external-link" href="https://www.chemaxon.com/marvin/help/developer/beans/api/com/chemaxon/mapper/AutoMapper.html" rel="nofollow"><code>com.chemaxon.mapper.Automapper</code></a></li>
            </ul></li>
           <li><strong>Structure Checker</strong>
            <ul>
             <li>&quot;Traditional nitrogen representation allowed&quot; option is available in Valence Error Checker. <a class="external-link" href="http://www.chemaxon.com/marvin/help/structurechecker/checkerlist.html#valenceerror" rel="nofollow">Documentation</a></li>
             <li><em>External structure checker and fixer manager</em> is integrated into Structure Checker GUI application. It is accessible from the preferences menu as &quot;Checker/Fixer Manager&quot;. <a class="external-link" href="http://www.chemaxon.com/marvin/help/structurechecker/structurechecker.html#externalchceckers" rel="nofollow">Documentation</a></li>
            </ul></li>
          </ul>
          <h4 id="id-6.0-Knownissue">Known issue</h4>
          <ul>
           <li><strong>MarvinSketch Applet</strong>
            <ul>
             <li>External checker classes do not load if browser uses Java SE 6. Please, use Java SE 7 for successful checker class loading.</li>
            </ul></li>
          </ul>
          <h4 id="id-6.0-Bugfixes.3">Bugfixes</h4>
          <ul>
           <li><strong>MarvinSketch Menu</strong>
            <ul>
             <li>The &quot;Recent Files&quot; list was not updated with the new file name when the file contained multiple structures and more records were selected.</li>
            </ul></li>
           <li><strong>Painting</strong>
            <ul>
             <li>There was a gap between the lines of the double bond and the single bonds when two single bonds connected to the same (non-visible) carbon atom on one end of a double bond.</li>
             <li>Electron flow arrows partially hid lone pairs in some cases.</li>
            </ul></li>
           <li><strong>Clipboard handling</strong>
            <ul>
             <li>When copying and pasting between Marvin applications in Linux, an exception was thrown which froze the application where the copy operation had been performed. The exception originated from the applied freehep library, which has been updated.</li>
            </ul></li>
           <li><strong>MarvinSpace</strong>
            <ul>
             <li>3D alignment: MCS mode is fixed to align molecules in 3D.</li>
            </ul></li>
           <li><strong>Image I/O</strong>
            <ul>
             <li>Image export did not remember the last folder used for saving.</li>
             <li>Molecule source was not saved in jpg, png, and svg files when &quot;Save As Image&quot; was selected. <a class="external-link" href="https://www.chemaxon.com/forum/ftopic10044.html" rel="nofollow">Forum topic</a></li>
            </ul></li>
           <li><strong>Import/Export</strong>
            <ul>
             <li><strong>MOL, SDF, RXN, RDF</strong>
              <ul>
               <li>Structures that contained two S-groups with exactly the same atoms imported from MDL MOL format could cause <code>java.lang.RuntimeException</code>. <a class="external-link" href="https://www.chemaxon.com/forum/ftopic10193.html" rel="nofollow">Forum topic</a></li>
               <li>Property name was imported as its value from SDF file when the property name contained square brackets.</li>
               <li>Exporting a peptide sequence containing bridge between two ungrouped amino acids to MOL format (V2000 or V3000), then importing it back, changed the coordinates of the contracted groups.</li>
              </ul></li>
             <li><strong> CDX/CDXML </strong>
              <ul>
               <li>CDX files created by ChemDraw 9.0.1 containing abbreviated groups with no attachment points was not imported.</li>
               <li>ChemDraw CDX files containing textboxes with Chinese characters were not imported.</li>
               <li>Chemical formula-like subscript of a Superatom S-group was read in reverse order from CDX when the group was to the left of the fragment.</li>
               <li>Superatom S-groups not having crossing bonds were read as textbox from CDX files.</li>
               <li>CDX files exported from ChemDraw 5.0 and containing contracted Superatom S-group was not imported.</li>
               <li>CDX import read half headed arrows as full headed.</li>
              </ul></li>
             <li><strong>CXSMILES/CXSMARTS</strong>
              <ul>
               <li>CXSMARTS export did not keep CIS/TRANS information for symmetrical structures.</li>
               <li>Extended part of CXSMARTS was not written correctly for any halogen query atoms.</li>
              </ul></li>
             <li><strong>SMILES/SMARTS</strong>
              <ul>
               <li>Unique SMILES string generation for a fragment could have differed depending on the fragment count of the molecule</li>
              </ul></li>
             <li><strong>Peptide</strong>
              <ul>
               <li>User defined custom amino acid dictionary caused wrong custom amino acid structure or <code>java.lang.ArrayIndexOutOfBoundsException</code> during peptide import.</li>
              </ul></li>
             <li><strong>Name to Structure (n2s)</strong>
              <ul>
               <li>When calling a custom name-to-structure webservice, names with special characters (including space) were not URL encoded, possibly resulting in failed conversions for such names.</li>
               <li>Name containing fused rings and converted to all uppercase letters were not recognized.</li>
               <li>Name to Structure could get stuck in a deadlock during the first usage when that occurred concurrently in several threads.</li>
               <li>When using the &quot;name converters&quot; API, converters were not properly removed after using thread-local converters.</li>
              </ul></li>
             <li><strong>Document to Structure (d2s)</strong>
              <ul>
               <li>When interrupted while doing OCR or OSR, document to structure logged an exception about the interrupt. This caused in particular a warning dialog to appear in Instant JChem during a document import.</li>
               <li>Names were not recognized when they contained a complex bracketed bond locant followed by a comma, such as &quot;2(7)&quot; in tricyclo[6.3.1.0<sup>2,7</sup>]dodeca-2(7),3,5-triene.</li>
               <li>Scrolling back in MarvinView could have caused failures for very large documents (more than 1000 structures) and names with non-ASCII characters.</li>
              </ul></li>
             <li><strong>Structure to Name(s2n)</strong>
              <ul>
               <li><code>name:source</code> returned an empty name for empty structures instead of the actual stored name.</li>
               <li>The name &quot;perimidine&quot; was not attributed to the right structure.</li>
              </ul></li>
            </ul></li>
           <li><strong> Molecule Representation</strong>
            <ul>
             <li><strong>Implicit/Explicit Hydrogen Conversion</strong>
              <ul>
               <li>Removing explicit hydrogens on some larger PubChem compounds caused infinite loop.</li>
              </ul></li>
            </ul></li>
           <li><strong>Clean 2D </strong>
            <ul>
             <li>Partial clean misplaced the molecule when only one atom was fixed.</li>
             <li>Clean 2D failed to preserve cis/trans stereo information in some cases.</li>
            </ul></li>
           <li><strong>Calculations</strong>
            <ul>
             <li><strong>NMR (HNMR, CNMR Prediction, ...)</strong>
              <ul>
               <li>PDF export of NMR spectra was failed when the generated new page contained a blank line only.</li>
               <li>When displaying NMR spectra of selected tautomers, the background of the tautomer preview window was black in Linux OS.</li>
              </ul></li>
            </ul></li>
          </ul>
          <h4 id="id-6.0-Deprecation">Deprecation</h4>
          <ul>
           <li><strong>Deprecated methods</strong>
            <ul>
             <li><strong>Molecule Representation </strong>
              <ul>
               <li>Valence Check: <code> MoleculeGraph.VALENCE_CHECK_AMBIGUOUS, MoleculeGraph.ValenceCheckState, MoleculeGraph.setValenceCheckState(ValenceCheckState), MoleculeGraph.getValenceCheckState() MoleculeGraph.getPossibleAttachmentPoints(MolAtom) </code></li>
               <li>S-groups: <code> SuperatomSgroup.getAttachAtoms(), SuperatomSgroup.getLegalAttachAtoms(), SuperatomSgroup.getFreeLegalAttachAtoms(), SuperatomSgroup.isFreeLegalAttachAtom(MolAtom), SuperatomSgroup.isLegalAttachment(MolAtom), SuperatomSgroup.sortXBonds(), MolAtom.getAttach(), MolAtom.setAttach(int), MolAtom.setAttach(int, Sgroup) </code></li>
               <li>Elemental Analysis: <code> ElementalAnalyser.setMolecule(SMolecule) </code></li>
               <li>Other deprecated methods <code> Molecule.exportToFormat(String), Molecule.exportToBinFormat(String), Molecule exportToObject(String) </code> throws <code>java.lang.IOException</code>.</li>
              </ul></li>
             <li><strong>Structure to Name (s2n)</strong>
              <ul>
               <li>The method <code><a class="external-link" href="http://chemaxon.marvin.io.formats.name" rel="nofollow">chemaxon.marvin.io.formats.name</a>.nameexport.IUPACNamer.generateName(Molecule)</code> , though not documented, was public for internal reasons, and used a non-thread-safe way to receive options. It is now deprecated in favor of <code>chemaxon.formats.MolExporter(Molecule, String format)</code> , where format can be for instance &quot;name&quot; for IUPAC name, &quot;name:t&quot; for traditional name. See <a class="external-link" href="https://www.chemaxon.com/marvin/help/formats/name-doc.html#export" rel="nofollow">https://www.chemaxon.com/marvin/help/formats/name-doc.html#export</a> for all possible options.</li>
              </ul></li>
             <li><strong>Automapper</strong>
              <ul>
               <li><code>chemaxon.marvin.modules.AutoMapper</code></li>
              </ul></li>
            </ul></li>
           <li><strong> Removed methods</strong>
            <ul>
             <li><strong>Molecule Representation</strong>
              <ul>
               <li><code> MolAtom.setMassnoIfKnown(String), MolAtom.isNobleGas(), MolAtom.isArrowEnd(), MolAtom.valenceCheck(), MolAtom.getEdgeCount(), MolAtom.getEdge(int), MolAtom.getEdgeTo(MolAtom), MolAtom.haveSimilarEdges(MolAtom), </code></li>
               <li><code> MolBond.ARROW(), MolBond.isCoordinative(), MolBond.isArrow(), MolBond.getNode1(), MolBond.getNode2(), MolBond.getOtherNode(MolAtom), MolBond.cloneEdge(MolAtom a1, MolAtom a2) </code></li>
               <li><code> MoleculeGraph.AROM_CHEMAXON, MoleculeGraph.AROM_DAYLIGHT(), MoleculeGraph.setSetSeqs(int id), MoleculeGraph.getAromrings(), MoleculeGraph.getNonAromrings(), MoleculeGraph.getAromrings(int), MoleculeGraph.getNonAromrings(int), MoleculeGraph.mergeFrags(), MoleculeGraph.getFragIds(), MoleculeGraph.getfindFragById(int fragId, MoleculeGraph frag), MoleculeGraph.getNode(int), MoleculeGraph.setNode(int, MolAtom), MoleculeGraph.insertNode(int, MolAtom), MoleculeGraph.removeNode(MolAtom), MoleculeGraph.removeNode(MolAtom, int), MoleculeGraph.removeNodeVector(), MoleculeGraph.getEdgeCount(), MoleculeGraph.getEdge(int), MoleculeGraph.setEdge(int, MolBond), MoleculeGraph.insertEdge(int, MolBond), MoleculeGraph.insertEdgeInOrder(MolBond, MolBond[]), MoleculeGraph.replaceEdge(MolBond, MolBond), MoleculeGraph.removeEdge(MolBond), MoleculeGraph.getEdgeArray(), MoleculeGraph.getEdgeVector(), MoleculeGraph.getAllEdges(), MoleculeGraph.regenEdges(), MoleculeGraph.sortEdgesAccordingTo(MolBond[]), MoleculeGraph.mergeNodes(MolAtom that, MolAtom), MoleculeGraph.mergeFrags(int, int) </code></li>
               <li><code> Molecule.mergeFrags()</code></li>
               <li><code> RgMolecule.setAbsStereo(boolean, int, int), RgMolecule.createMol(String), RgMolecule.getBtab() </code></li>
               <li><code> RxnMolecule.getBtab(), RxnMolecule.getStructureCount(int), RxnMolecule.getStructure(int, int), RxnMolecule.addStructure(Molecule, int), RxnMolecule.removeStructure(int, int) </code></li>
               <li><code> CEdge, CNode, CGraph </code></li>
               <li><code> MDocument.setGUIProperyContainer(MPropertyContainer) </code></li>
              </ul></li>
            </ul></li>
          </ul>
         </div> 
        </div> 
       </div> 
      </div>
      <p></p> 
     </div> 
    </div> 
   </div> 
   <div id="footer"> 
    <section class="footer-body" style="background-image:none"> 
     <p style="font-size:16px">&copy; ChemAxon Ltd.</p> 
    </section> 
   </div> 
  </div>   
 </body>
</html>