<!DOCTYPE html>
<html>
 <head> 
  <title>MarvinSketch : 6.3</title> 
  <link rel="stylesheet" href="styles/site.css" type="text/css" /> 
  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" /> 
  <style>
#content {
    -moz-box-sizing: border-box;
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    clear: both;
    position: relative;
    background: #FFFFFF !important;
}

.aui-page-panel {
    background: #FFFFFF !important;
    border: 1px solid #ccc;
    border-left-width: 0;
    border-right-width: 0;
    -moz-box-sizing: border-box;
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    clear: both;
    display: block;
    margin: 20px 0 0 0;
    padding: 0;
    position: relative;
    width: 100%;
}
.message-content {
	border: solid 2px #fee796;
	background-color: #fefaea;
	border-radius: 10px;
	display: table;
	padding: 4px;
	margin: 4px 4px 0px 4px;
}
</style> 
 </head> 
 <body class="theme-default aui-theme-default"> 
  <div id="page"> 
   <div id="main" class="aui-page-panel"> 
    <div id="main-header"> 
     <h1 id="title-heading" class="pagetitle"> <span id="title-text"> MarvinSketch : 6.3 </span> </h1> 
    </div> 
    <div id="content" class="view"> 
     <div id="main-content" class="wiki-content group"> 
      <p></p>
      <div class="contentLayout2"> 
       <div class="columnLayout three-equal" data-layout="three-equal"> 
        <div class="cell normal" data-type="normal"> 
         <div class="innerCell"> 
          <div class="panel" style="border-width: 1px;">
           <div class="panelContent"> 
            <p><style type="text/css">/*<![CDATA[*/
div.rbtoc1439201040622 {padding: 0px;}
div.rbtoc1439201040622 ul {list-style: disc;margin-left: 0px;}
div.rbtoc1439201040622 li {margin-left: 0px;padding-left: 0px;}

/*]]>*/</style></p>
            <div class="toc-macro rbtoc1439201040622"> 
             <ul class="toc-indentation"> 
              <li><a href="https://docs.chemaxon.com/display/products/6.3#id-6.3-September16th,2014:Marvin6.3.4.">September 16th, 2014: Marvin 6.3.4.</a></li> 
              <li><a href="https://docs.chemaxon.com/display/products/6.3#id-6.3-August5th,2014:Marvin6.3.3.">August 5th, 2014: Marvin 6.3.3.</a> 
               <ul class="toc-indentation"> 
                <li><a href="https://docs.chemaxon.com/display/products/6.3#id-6.3-Bugfixes">Bugfixes</a></li> 
               </ul> </li> 
              <li><a href="https://docs.chemaxon.com/display/products/6.3#id-6.3-July7th,2014:Marvin6.3.2">July 7th, 2014: Marvin 6.3.2</a> 
               <ul class="toc-indentation"> 
                <li><a href="https://docs.chemaxon.com/display/products/6.3#id-6.3-Bugfixes.1">Bugfixes</a></li> 
               </ul> </li> 
              <li><a href="https://docs.chemaxon.com/display/products/6.3#id-6.3-June4th,2014:Marvin6.3.1">June 4th, 2014: Marvin 6.3.1</a> 
               <ul class="toc-indentation"> 
                <li><a href="https://docs.chemaxon.com/display/products/6.3#id-6.3-Newfeaturesandimprovements">New features and improvements</a></li> 
                <li><a href="https://docs.chemaxon.com/display/products/6.3#id-6.3-Bugfixes.2">Bugfixes</a></li> 
               </ul> </li> 
              <li><a href="https://docs.chemaxon.com/display/products/6.3#id-6.3-April18th,2014:Marvin6.3.0">April 18th, 2014: Marvin 6.3.0</a> 
               <ul class="toc-indentation"> 
                <li><a href="https://docs.chemaxon.com/display/products/6.3#id-6.3-Newfeaturesandimprovements.1">New features and improvements</a></li> 
                <li><a href="https://docs.chemaxon.com/display/products/6.3#id-6.3-Bugfixes.3">Bugfixes</a></li> 
                <li><a href="https://docs.chemaxon.com/display/products/6.3#id-6.3-Deprecatedandremovedmethods">Deprecated and removed methods</a></li> 
                <li><a href="https://docs.chemaxon.com/display/products/6.3#id-6.3-Knownissue">Known issue</a></li> 
               </ul> </li> 
             </ul> 
            </div>
            <p></p> 
           </div>
          </div>
         </div> 
        </div> 
        <div class="cell normal" data-type="normal"> 
         <div class="innerCell"> 
          <p>&nbsp;</p>
         </div> 
        </div> 
        <div class="cell normal" data-type="normal"> 
         <div class="innerCell"> 
          <p>&nbsp;</p>
         </div> 
        </div> 
       </div> 
       <div class="columnLayout single" data-layout="single"> 
        <div class="cell normal" data-type="normal"> 
         <div class="innerCell"> 
          <h3 id="id-6.3-September16th,2014:Marvin6.3.4.">September 16th, 2014: Marvin 6.3.4.</h3>
          <p style="margin-left: 60.0px;">No changes.</p>
          <h3 id="id-6.3-August5th,2014:Marvin6.3.3.">August 5th, 2014: Marvin 6.3.3.</h3>
          <h4 id="id-6.3-Bugfixes"><span>Bugfixes</span></h4>
          <ul>
           <li><strong>Import/Export</strong>
            <ul>
             <li><span><span>OLE objects created by Accelrys Draw 4.1 and further versions could not be redirected by MarvinOLE server.</span></span></li>
             <li>OLE opened in 3D rotation mode (F7) mode.</li>
            </ul></li>
          </ul>
          <h3 id="id-6.3-July7th,2014:Marvin6.3.2">July 7th, 2014: Marvin 6.3.2</h3>
          <h4 id="id-6.3-Bugfixes.1">Bugfixes</h4>
          <ul>
           <li>&nbsp;&nbsp;&nbsp;<strong> Applet</strong>
            <ul>
             <li>Communication problem involving JavaScript was fixed.</li>
            </ul></li>
          </ul>
          <h3 id="id-6.3-June4th,2014:Marvin6.3.1">June 4th, 2014: Marvin 6.3.1</h3>
          <h4 id="id-6.3-Newfeaturesandimprovements">New features and improvements</h4>
          <ul>
           <li><strong> Image I/O </strong>
            <ul>
             <ul>
              <li>New image IO parameter &quot;aminoAcidBondColoringEnabled&quot; has been introduced for peptide bond coloring.</li>
             </ul>
            </ul></li>
           <li><strong> Import/Export </strong>
            <ul>
             <li><strong> SKC </strong>
              <ul>
               <li>If they are set to visible in ISIS, &quot;Atom numbers&quot; are imported as atom mapping from SKC.</li>
              </ul></li>
            </ul></li>
          </ul>
          <h4 id="id-6.3-Bugfixes.2">Bugfixes</h4>
          <ul>
           <li><strong>Clipboard handling</strong>
            <ul>
             <ul>
              <li>If a structure had been pasted onto an empty canvas and MarvinSketch was closed without modifying the structure, the structure was not saved.</li>
             </ul>
             <ul>
              <li>Files with &quot;smiles&quot; extension did not load by dragging them and dropping into MarvinSketch.</li>
             </ul>
             <ul>
              <li>Clicking on the Transfer button (of MarvinSketch) after editing a structure in MarvinView caused NPE.</li>
             </ul>
            </ul></li>
           <li><strong>Editing</strong>
            <ul>
             <ul>
              <li>If a template was merged to an atom of an abbreviated group which did not have any attachment point, the abbreviated group was not ungrouped.</li>
             </ul>
             <ul>
              <li>Bond shortcuts did not work until the user pressed the &quot;Bond&quot; button on the &quot;Tools&quot; toolbar.</li>
             </ul>
            </ul></li>
           <li><strong> Image I/O </strong>
            <ul>
             <ul>
              <li>Bond set coloring image export parameter was not effective, and in some cases it was able to modify the atom set coloring.</li>
             </ul>
             <ul>
              <li>When saving a structure in pdf format, the image was cut too narrow.</li>
             </ul>
             <ul>
              <li>Circled charge was not copied into PDF clipboard format on Mac. Currently it is represented without the circle.</li>
             </ul>
            </ul></li>
           <li><strong> Molecule Representation </strong>
            <ul>
             <li><strong> Valence Check </strong>
              <ul>
               <li>False valence error appeared on aromatic rings containing Nitrogen-like atoms in their aromatic form.</li>
              </ul></li>
             <li><strong> S-groups </strong>
              <ul>
               <li>The direction of S-group labels was changed at wrong angle.</li>
              </ul></li>
            </ul></li>
           <li><strong>Painting </strong>
            <ul>
             <ul>
              <li>The &quot; * &quot; character was not supported in (among others) Arial, Times New Roman, and Lucida font family.</li>
             </ul>
             <ul>
              <li>The leftname/rightname rendering of abbreviated groups have been incorrect in several cases such as &quot;OMe&quot;.</li>
             </ul>
            </ul></li>
           <li><strong>Name to Structure (n2s)</strong>
            <ul>
             <li>Some substituents separated by a space from an acid parent were wrongly interpreted as esters.</li>
             <li>The naming webservice was not working in the applet.</li>
            </ul></li>
          </ul>
          <h3 id="id-6.3-April18th,2014:Marvin6.3.0">April 18th, 2014: Marvin 6.3.0</h3>
          <p><a href="https://docs.chemaxon.com/display/products/History+of+Changes+-+Marvin+JS" rel="nofollow">Marvin JS</a> 6.3.0 history of changes</p>
          <h4 id="id-6.3-Newfeaturesandimprovements.1">New features and improvements</h4>
          <ul>
           <li><strong>Painting</strong>
            <ul>
             <li>Bond properties are displayed on the canvas.</li>
            </ul></li>
           <li><strong>Clipboard handling</strong>
            <ul>
             <li>When the clipboard format is not recognized as a chemical object, the format chooser dialog is not displayed and Marvin does not import anything. <a class="external-link" href="https://www.chemaxon.com/forum/ftopic11673.html" rel="nofollow">Forum</a></li>
            </ul></li>
           <li><strong>MarvinSketch Menu</strong>
            <ul>
             <li>The &quot;Switch Transformation&quot; menu item is removed from the &quot;Edit &gt; Transform&quot; menu.</li>
             <li>The &quot;Graph Invariants&quot; menu option is removed from the &quot;View &gt; Advanced&quot; menu.</li>
            </ul></li>
           <li><strong>MarvinSketch Dialogs</strong>
            <ul>
             <li>Look&amp;Feel option has been eliminated from the &quot;Preferences&quot; dialog.</li>
            </ul></li>
           <li><strong> Import/Export </strong>
            <ul>
             <li><strong> CXSMILES, CXSMARTS </strong>
              <ul>
               <li>R-logic is supported by the CXSMILES and CXSMARTS formats.</li>
               <li>Atom properties of homology groups are supported by the CXSMILES/CXSMARTS formats.</li>
              </ul></li>
             <li><strong> CDX, CDXML </strong>
              <ul>
               <li>3-letter abbreviated amino acid labels can be imported from CDX. <a class="external-link" href="https://www.chemaxon.com/forum/ftopic54008.html" rel="nofollow">Forum</a></li>
              </ul></li>
             <li><strong>Name to Structure (n2s)</strong>
              <ul>
               <li>Japanese Name to Structure is supported. It requires a separate license.</li>
               <li>The custom dictionary can be encoded in UTF-16 (both UTF-16BE and UTF-16LE variants) even if the system default encoding is different. The file needs to start with a byte order mark (BOM) character.</li>
              </ul></li>
             <li><strong>Document to Structure (d2s)</strong>
              <ul>
               <li>OSRA 2.0 is supported.</li>
               <li>A format option has been introduced allowing the user to select an optical structure recognition tool: CLiDE, OSRA or Imago when more than one is installed on the computer. For instance, <code>d2s:osra</code> will request OSRA to be used.</li>
              </ul></li>
             <li><strong>Structure to Name (s2n)</strong>
              <ul>
               <li>Locants with a number and an atom type are written, for instance, as N6 instead of 6-N following the latest IUPAC recommendations.</li>
              </ul></li>
            </ul></li>
           <li><strong> Molecule Representation </strong>
            <ul>
             <li>Method <code>chemaxon.struc.MolBond.equals() </code> compares the bonds by reference and not by the end points.</li>
             <li><strong> Clean 2D </strong>
              <ul>
               <li>Clean 2D of structures containing explicit hydrogen atoms is improved. <a class="external-link" href="https://www.chemaxon.com/forum/ftopic11342.html" rel="nofollow">Forum</a></li>
              </ul></li>
            </ul></li>
           <li><strong>Applet</strong>
            <ul>
             <li>When the Marvin Applet does not work because the Java version is outdated, a proper warning message is displayed.</li>
            </ul></li>
           <li><strong> Licensing</strong>
            <ul>
             <li>Reading of licenses has changed: now licenses in the ChemAxon home directory are read even if the <code>CHEMAXON_LICENSE_URL</code> environment variable or the <code>chemaxon.license.url</code> java system property is set.</li>
            </ul></li>
           <li><strong>Structure Checker</strong>
            <ul>
             <li>&quot;Double Bond Stereo Error Checker&quot; has been introduced, which searches for invalid either double bonds. Its corresponding fixer converts the invalid either double bonds to plain double bonds. <a class="external-link" href="http://www.chemaxon.com/marvin/help/structurechecker/checkerlist.html#doublebondstereo" rel="nofollow">Documentation</a></li>
             <li>&quot;Stereo Inversion Retention Mark Checker&quot; has been introduced, which identifies stereo inversion/retention marks on single molecules and in reactions. Its associated fixer removes the stereo inversion/retention marks from the structures. <a class="external-link" href="http://www.chemaxon.com/marvin/help/structurechecker/checkerlist.html#stereoinversionretentionmark" rel="nofollow">Documentation</a></li>
             <li>&quot;Valence Error Checker&quot; detects all valence errors in Markush structures, even if the valence error is inside R-group definitions. Its fixer corrects all fixable valence errors in Markush structures.</li>
            </ul></li>
           <li><strong>Calculations</strong>
            <ul>
             <li><strong>Solubility</strong>
              <ul>
               <li>Aqueous Solubility Predictor has been added to Calculator Plugins. <a class="external-link" href="http://www.chemaxon.com/marvin/help/calculations/solubility.html" rel="nofollow">Documentation</a></li>
              </ul></li>
             <li><strong>Stereo Analysis</strong>
              <ul>
               <li>Available via cxcalc and Chemical Terms. <a class="external-link" href="http://www.chemaxon.com/marvin/help/calculations/stereoanal.html" rel="nofollow">Documentation</a></li>
              </ul></li>
             <li><strong>Conformations</strong>
              <ul>
               <li>Advanced conformational search. <a class="external-link" href="http://www.chemaxon.com/marvin/help/calculations/conformation.html#conformation" rel="nofollow">Documentation</a></li>
              </ul></li>
            </ul></li>
          </ul>
          <h4 id="id-6.3-Bugfixes.3">Bugfixes</h4>
          <ul>
           <li><strong>Editing</strong>
            <ul>
             <li>Trying to merge an S-group on a multicenter atom threw a <code>NullPointerException</code>.</li>
             <li>Hovering over a rounded rectangle with the mouse cursor caused a <code>NullPointerException</code>.</li>
             <li>Replacing a selected single atom with a hydrogen using the toolbar button resulted in a &quot;HH&quot; atom.</li>
             <li>Bonds could not be drawn from hydrogen atoms to other atoms.</li>
             <li>The &quot;Absolute&quot; chiral flag disappeared from the canvas when the reaction arrow was deleted.</li>
             <li>The brackets of an S-group disappeared when the group became part of a reaction.</li>
             <li>Replacing an amino acid in a sequence merged the new amino acid in the wrong direction.</li>
             <li>When single atoms in a chain were replaced by amino acids, attachment orders were incorrect.</li>
             <li>Rotation of molecules in 3D did not work properly.</li>
             <li>If a query property was set for an atom and - in a later step- a bond was drawn from this atom, the value of the atomic query property changed to zero regardless from its original setting.</li>
            </ul></li>
           <li><strong>Load/Save handling in UI</strong>
            <ul>
             <li>Changing the background color when a 3D structure was converted to 2D, then saving in PNG format saved the structure with the earlier, 3D background color.</li>
            </ul></li>
           <li><strong>Clipboard handling</strong>
            <ul>
             <li>Structure display options were not preserved when structures were copy-pasted as Marvin OLE objects.</li>
             <li>Pasting a file in a one-line molecule format (<em>e.g.</em>, SMILES) on the canvas merged the lines of the file content. Currently, only the first line of the file is pasted.</li>
            </ul></li>
           <li><strong>MarvinView GUI</strong>
            <ul>
             <li>The &quot;Save Selection...&quot; menu option could not be used for selecting structures for saving from a multimolecule file.</li>
            </ul></li>
           <li><strong> Import/Export </strong>
            <ul>
             <li><strong> MRV, CML </strong>
              <ul>
               <li>CML and MRV import threw <code>java.lang.ArrayIndexOutOfBoundsException</code> when reading parallel bonds atoms. Now <code>chemaxon.formats.MolFormatException </code> is thrown instead.</li>
              </ul></li>
             <li><strong> SMILES, SMARTS </strong>
              <ul>
               <li>SMARTS string starting with parentheses was recognized as SMILES.</li>
              </ul></li>
             <li><strong> CXSMILES, CXSMARTS </strong>
              <ul>
               <li>Chiral flag (or molecular absolute stereo) was not exported to CXSMILES if enhanced stereo existed in the molecule.</li>
              </ul></li>
             <li><strong> InChi, InChiKey </strong>
              <ul>
               <li>Coordinate bond was exported as single bond to InChi. From now, we do not export these bonds according to the IUPAC recommendation.</li>
              </ul></li>
             <li><strong>Name to Structure (n2s)</strong>
              <ul>
               <li>The multiplying prefix &quot;eicosa&quot;, a non-IUPAC variant of &quot;icosa&quot; was not supported.</li>
               <li>The Chinese names of some alkanoic acids, such as ????, were not supported.</li>
               <li>Some common names containing &quot;chloride&quot; were imported without a chlorine atom.</li>
               <li>Some chiral heterocenters were reversed in anhydrides.</li>
               <li>Some names with <em>cis</em> and <em>trans</em> stereodescriptors for rings were imported without stereochemistry.</li>
               <li>Entries in the custom dictionary that consist of an empty structure failed to be imported.</li>
               <li>The stereochemistry of some sugars, for example, 1-(2-Deoxy-2,2-difluoro-Î²-L-erythro-pentofuranos-1-yl)uracil, was wrong.</li>
               <li>Stereochemistry was missing on some chiral centers of steroid derivatives.</li>
              </ul></li>
             <li><strong>Document to Structure (d2s)</strong>
              <ul>
               <li>The encoding of HTML files was not always detected from the META tag.</li>
               <li>When the extraction of OLE embedded structures from Office documents was disabled using the <code>d2s:-ole</code> format option, the optical structure recognition of images was disabled as well.</li>
               <li>The character position field of structures was sometimes higher by a few characters when extracted from HTML documents with CRLF (\r\n) line endings.</li>
               <li>The CLiDE optical structure recognition tool was not automatically detected on 64 bit versions of Windows.</li>
               <li>When using OSRA, some structures with implicit hydrogens were wrongly filtered out.</li>
               <li>After processing only a part of an HTML document, the processing of the next HTML document could give some incorrect results.</li>
              </ul></li>
             <li><strong>Structure to Name (s2n)</strong>
              <ul>
               <li>IUPAC numbering was not displayed for some classes of structures, for instance, secondary amines.</li>
               <li>The common names generated for some structures with a chloride counter-ion did not include the ending &quot;chloride&quot;.</li>
               <li>In a few cases, generated traditional names were imperfect, for instance, &quot;nitrobenzeen&quot; instead of &quot;nitrobenzene&quot;.</li>
              </ul></li>
            </ul></li>
           <li><strong> Molecule Representation </strong>
            <ul>
             <li><strong> Clean 2D </strong>
              <ul>
               <li>Clean failed on atoms having two chain ligands with the same length.</li>
              </ul></li>
             <li><strong> Implicit/Explicit Hydrogen Conversion </strong>
              <ul>
               <li>Hydrogen atoms were added with invalid coordinates to carbon atoms having triple bond in 3D. <a class="external-link" href="https://www.chemaxon.com/forum/ftopic8593.html" rel="nofollow">Forum</a></li>
              </ul></li>
            </ul></li>
           <li><strong>Image I/O</strong>
            <ul>
             <li><strong><em>R/S</em></strong> stereo labels on chiral atoms were partly covered by bonds during EMF export. <a class="external-link" href="https://www.chemaxon.com/forum/ftopic6964.html&amp;highlight=" rel="nofollow">Forum</a></li>
            </ul></li>
           <li><strong>Structure Checker</strong>
            <ul>
             <li>&quot;Bond Angle Error Checker&quot; reported incorrectly for molecules containing S-groups.</li>
             <li>The first letter of a typed word did not appear in the search box of the &quot;Configure Checkers&quot; panel.</li>
            </ul></li>
           <li><strong>Calculations</strong>
            <ul>
             <li>Remove Predictor from Marvin GUI, Chem Term, cxcalc and cxtrain.</li>
             <li><strong>Conformers</strong>
              <ul>
               <li>Lowest energy conformer calculation has been fixed.</li>
               <li>Marvin - Conformers Plugin options has been improved along with corresponding documentation.</li>
              </ul></li>
            </ul></li>
          </ul>
          <h4 id="id-6.3-Deprecatedandremovedmethods">Deprecated and removed methods</h4>
          <ul>
           <li><a href="https://docs.chemaxon.com/display/SR/Methods+Deprecated+or+Removed+in+version+6.3" rel="nofollow">Deprecated and removed methods</a></li>
          </ul>
          <h4 id="id-6.3-Knownissue">Known issue</h4>
          <ul>
           <li><strong>Structure Checker</strong>
            <ul>
             <li>Online documentation help link does not work on Linux OS.</li>
            </ul></li>
          </ul>
         </div> 
        </div> 
       </div> 
      </div>
      <p></p> 
     </div> 
    </div> 
   </div> 
   <div id="footer"> 
    <section class="footer-body" style="background-image:none"> 
     <p style="font-size:16px">&copy; ChemAxon Ltd.</p> 
    </section> 
   </div> 
  </div>   
 </body>
</html>